presto-verifier/src/main/java/com/facebook/presto/verifier/event/DeterminismAnalysisRun.java

```diff
@@ -85,6 +107,20 @@ public Builder setQueryId(String queryId)
             return this;
         }
 
+        public Builder addSetupQueryId(String queryId)
+        {
+            checkState(!this.queryId.isPresent(), "queryId is already set");
```

Comment:
is this the right check?

Reply:
Good catch. Remove those checks and added unit test coverage.